src/hotspot/share/classfile/moduleEntry.hpp

```diff
@@ -68,7 +68,9 @@ class ModuleEntry : public CHeapObj<mtModule> {
                                        // for shared classes from this module
   Symbol*          _name;              // name of this module
   ClassLoaderData* _loader_data;
-  GrowableArray<ModuleEntry*>* _reads; // list of modules that are readable by this module
+  void* _reads; // list of modules that are readable by this module
+  DEBUG_ONLY(bool _reads_is_growable);
+  DEBUG_ONLY(bool _reads_is_archived);
```

Comment:
Do we really need two bool members?  I've not delved deeply, but it looks like one bool flag would
be sufficient, so long as one is okay with initializing as having a null GA to start with.  This would
also avoid any questions about what it means for both to be true (shouldn't be possible, but bugs
happen, and the new state checks don't catch that) or both false.

Reply:
Is making this state information debug-only the best choice?  There were a couple of places where
I wondered if things might be simpler if we could test the state, rather than only asserting it.  If worried
about the additional space, there is some unused alignment padding after `_is_patched`.  Of course,
this would require managing the state in product builds, so it's not entirely free.